builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -795,6 +795,19 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 			networkInterfaces = append(networkInterfaces, networkInterface)
 		}
 	}
+	for _, v := range mvm.Guest.IpStack {
+		for _, route := range v.IpRouteConfig.IpRoute {
```

Comment:
Yes. I think we need some more checks here. `Guest`, `IpStack`, `IpRouteConfig` and `IpRoute` could be `nil`